packages/main/cypress/specs/TableSelection.cy.tsx

```diff
@@ -0,0 +1,374 @@
+					"checkbox": false,
+				},
+				"row": {
+					"exists": true,
+					"checkbox": true,
+				},
+			},
+			"SPACE": {
+				"space_0": "0",
+				"space_4": "4",
+			},
+			"ARROWS_BOX": {
+				"arrow_initial": "0",
+				"arrow_down": "1",
+				"arrow_up": "0",
+			},
+			"MOUSE": {
+				"mouse_0": "0",
+				"mouse_4": "4",
+			},
+			"RANGE_MOUSE": {
+				"range_mouse_initial": "0",
+				"range_mouse_final": "4",
+				"range_mouse_edge": "0",
+			},
+			"RANGE_KEYBOARD": {
+				"initial": "0",
+				"block_1": "0",
+				"block_2": "6",
+			},
+		},
+	},
+	"Multiple": {
+		"config": {
+			"mode": "Multiple",
+		},
+		"cases": {
+			"BOXES": {
+				"header": {
+					"exists": true,
+					"checkbox": true
+				},
+				"row": {
+					"exists": true,
+					"checkbox": true
+				}
+			},
+			"SPACE": {
+				"space_0": "0",
+				"space_4": "0 4"
+			},
+			"ARROWS_BOX": {
+				"arrow_initial": "0",
+				"arrow_down": "0",
+				"arrow_up": "0"
+			},
+			"MOUSE": {
+				"mouse_0": "0",
+				"mouse_4": "0 4"
+			},
+			"RANGE_MOUSE": {
+				"range_mouse_initial": "0",
+				"range_mouse_final": "0 1 2 3 4",
+				"range_mouse_edge": "0 1 2 3 4"
+			},
+			"RANGE_KEYBOARD": {
+				"initial": "0",
+				"block_1": "0 1 2 3 4",
+				"block_2": "0 1 2 3 4 6 7 8 9"
+			}
+		}
+	}
+};
+
+Object.entries(testConfig).forEach(([mode, testConfigEntry]) => {
+	describe(`Mode - ${mode}`, () => {
+		beforeEach(() => {
+			clearSelection();
+			mountTestpage();
+			changeMode(testConfigEntry.config.mode as TableSelectionMode);
+		});
+
+		it("Correct boxes are shown", () => {
+			cy.get("#table0").as("table");
+			cy.get("#table0").children("ui5-table-header-row").as("headerRow");
+			cy.get("#table0").children("ui5-table-row").get("[row-key=\"0\"]").as("row0");
+
+			cy.get("@table").should("exist");
+			cy.get("@headerRow").should("exist");
+			cy.get("@row0").should("exist");
+
+			cy.get("@headerRow").first().shadow().find("#selection-cell")
+				.should(testConfigEntry.cases.BOXES.header.exists ? "exist" : "not.exist");
+			cy.get("@row0").shadow().find("#selection-cell")
+				.should(testConfigEntry.cases.BOXES.row.exists ? "exist" : "not.exist");
+
+			cy.get("@headerRow").first().shadow().find("#selection-component")
+				.should(testConfigEntry.cases.BOXES.header.checkbox ? "exist" : "not.exist");
+			cy.get("@row0").shadow().find("#selection-component")
+				.should(testConfigEntry.cases.BOXES.row.checkbox ? "exist" : "not.exist");
+		});
+
+		it("select row via SPACE", () => {
+			cy.get("#table0").as("table");
+			cy.get("#table0").children("ui5-table-row").get("[row-key=\"0\"]").as("row0");
+			cy.get("#table0").children("ui5-table-row").get("[row-key=\"4\"]").as("row4");
+			cy.get("#selection").as("selection");
+
+			cy.get("@row0").realClick({ position: "left" });
+			cy.get("@row0").realPress("Space");
+			cy.get("[ui5-table-selection]").then(sel => {
+				expect(sel.get(0).getAttribute("selected")).to.equal(testConfigEntry.cases.SPACE.space_0);
```

Comment:
this problem isn't solved yet? maybe add a comment why you handle via JS API instead of cypress API and then we can maybe ask Sofia colleagues to have a look?